server/configuration.go

```diff
@@ -1,47 +1,148 @@
 package server
 
 import (
+	"errors"
 	"fmt"
+
+	"log"
 	"os"
+	"path/filepath"
 	"strings"
+
+	"github.com/dmarro89/dare-db/logger"
+	"github.com/dmarro89/dare-db/utils"
+
+	"github.com/spf13/viper"
 )
 
-const DARE_PORT = "DARE_PORT"
-const DARE_HOST = "DARE_HOST"
-const DARE_TLS_ENABLED = "DARE_TLS_ENABLED"
-const DARE_TLS_CERT_PRIVATE  = "DARE_TLS_CERT_PRIVATE"
-const DARE_TLS_CERT_PUBLIC = "DARE_TLS_CERT_PUBLIC"
-
-type Configuration struct {
-	Port        string
-	Host        string
-	TLSEnabled  bool
-	TLSCertFile string
-	TLSKeyFile  string
-}
-
-func NewConfiguration() *Configuration {
-	return &Configuration{
-		Port:        fmt.Sprintf("%s", getEnvOrDefault(DARE_PORT, "2605")),
-		Host:        fmt.Sprintf("%s", getEnvOrDefault(DARE_HOST, "")),
-		TLSEnabled:  getEnvBooleanOrDefault(DARE_TLS_ENABLED, false),
-		TLSCertFile: fmt.Sprintf("%s", getEnvOrDefault(DARE_TLS_CERT_PRIVATE , "")),
-		TLSKeyFile:  fmt.Sprintf("%s", getEnvOrDefault(DARE_TLS_CERT_PUBLIC, "")),
+var mapsEnvsToConfig map[string]string = make(map[string]string)
+
+func checkFileExists(filePath string) bool {
+	_, error := os.Stat(filePath)
+	return !errors.Is(error, os.ErrNotExist)
+}
+
+func createDirectory(dirPath string) {
+	_, err := os.Stat(dirPath)
+	if err != nil && os.IsNotExist(err) {
```

Comment:
I'd like to avoid if/else chains.
createDirectory can return an error and break the flow with a return error or return nil in case there is an error or not.

Reply:
The idea was to handle all edge cases in the function directly. Otherwise, these case should be handled everything time on return.

Reply:
Something like:
`func createDirectory(dirPath string) {
	err := os.MkdirAll(dirPath, 0755)
	switch {
	case err == nil:
		logger.Info("Directory created successfully:", dirPath)
	case os.IsExist(err):
		logger.Debug("Directory already exists: ", dirPath)
	default:
		logger.Error("Error creating directory:", err)
	}
}`

Reply:
Done